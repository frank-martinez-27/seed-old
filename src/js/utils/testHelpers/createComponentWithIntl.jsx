// https://github.com/yahoo/react-intl/blob/master/examples/jest-snapshot-testing/src/utils/createComponentWithIntl.js

import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line
import { IntlProvider } from 'react-intl';
import content from '../../constants/content.json';
import { getMessagesForLanguage } from '../../containers/Localization';
import { LANGUAGES } from '../../constants';

const createComponentWithIntl = (children, props = { locale: LANGUAGES.ENGLISH, messages: {} }) => {
  const translation = getMessagesForLanguage(props.locale, content.messages);
  return renderer.create(
    <IntlProvider {...props} messages={translation}>
      {children}
    </IntlProvider>
  );
};

export default createComponentWithIntl;
