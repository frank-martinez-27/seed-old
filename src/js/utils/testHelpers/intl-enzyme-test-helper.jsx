/**
 * https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#helper-function-1
 *
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme'; // eslint-disable-line import/no-extraneous-dependencies
import { LANGUAGES } from '../../constants';

const content = require('../../constants/content.json');

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider(
  { locale: LANGUAGES.ENGLISH, messages: content.messages.en },
  {}
);
const { intl } = intlProvider.getChildContext();

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

export function shallowWithIntl(node, { context } = {}) {
  return shallow(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
  });
}

export function mountWithIntl(node, { context, childContextTypes } = {}) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    childContextTypes: Object.assign({}, { intl: intlShape }, childContextTypes),
  });
}

/*
 * Repeatedly render a component tree using enzyme.shallow() until
 * finding and rendering TargetComponent.
 *
 * The `componentInstance` parameter is a React component instance.
 * Example: <MyComponent {...props} />
 *
 * The `TargetComponent` parameter is the React class (or function) that
 * you want to retrieve from the component tree.
 */
export function shallowUntilTarget(
  componentInstance,
  TargetComponent,
  { maxTries = 10, shallowOptions, _shallow = shallow } = {}
) {
  if (!componentInstance) {
    throw new Error('componentInstance parameter is required');
  }
  if (!TargetComponent) {
    throw new Error('TargetComponent parameter is required');
  }

  let root = _shallow(componentInstance, shallowOptions);

  if (typeof root.type() === 'string') {
    // If type() is a string then it's a DOM Node.
    // If it were wrapped, it would be a React component.
    throw new Error('Cannot unwrap this component because it is not wrapped');
  }

  for (let tries = 1; tries <= maxTries; tries++) {
    if (root.is(TargetComponent)) {
      // Now that we found the target component, render it.
      return root.shallow(shallowOptions);
    }
    // Unwrap the next component in the hierarchy.
    root = root.dive();
  }

  throw new Error(
    `Could not find ${TargetComponent} in rendered instance: ${componentInstance}; gave up after ${maxTries} tries`
  );
}
