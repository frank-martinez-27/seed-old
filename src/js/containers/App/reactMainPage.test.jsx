import React from 'react';
import { shallow } from 'enzyme';
import ReactMain from './reactMainPage';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import createComponentWithIntl from '../../utils/testHelpers/createComponentWithIntl';
import { shallowWithIntl } from '../../utils/testHelpers/intl-enzyme-test-helper';
import { MemoryRouter } from 'react-router-dom';

const MainPageComponent = (
  <IntlProvider locale="EN">
    <ReactMain />
  </IntlProvider>
);
describe('Main Page', () => {
  it('renders correctly', () => {
    const reactMainPage = createComponentWithIntl(
      <MemoryRouter>
        <ReactMain />
      </MemoryRouter>
    ).toJSON();
    expect(reactMainPage).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    shallow(<ReactMain />);
  });

  it('renders welcome message', () => {
    const wrapper = shallowWithIntl(<ReactMain />);
    const welcome = (
      <h1 className="App-title" data-testid="title">
        Welcome to React
      </h1>
    );
    expect(wrapper.find('[data-testid="title"]')).toBeDefined();
    expect(wrapper.contains(welcome)).toEqual(true);
    expect(wrapper).toContainReact(welcome);
  });

  it('e2e-like test - renders welcome message', () => {
    const { getByText } = render(MainPageComponent);
    expect(getByText('Welcome to React')).toBeInTheDOM();
    expect(getByText('Welcome to React')).toBeInTheDocument();
    expect(document.querySelector('[data-testid="header"]')).toContainElement(
      document.querySelector('[data-testid="title"]')
    );
  });
});
