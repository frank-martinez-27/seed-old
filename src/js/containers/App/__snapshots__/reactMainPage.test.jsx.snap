// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
    data-testid="header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
      data-testid="title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
    <span
      aria-hidden="true"
      className="icon icon-newspaper"
    />
  </p>
  <button
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      Hello World
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </button>
</div>
`;
